---
title: "NRG-GY003 WES Analysis"
subtitle: "Whole Exome genomic analysis of NRG-GY003 ovarian cancer tumors"
description: "Test"

author:
  - name: Kelsey Monson
    degrees:
      - PhD
      - MS
    orcid: 0000-0003-1093-8628
    email: kelsey.monson@mssm.edu
    affiliations:
      - ref: ISMMS

affiliations:
  - id: ISMMS
    name: Icahn School of Medicine at Mount Sinai
    city: New York
    state: NY
    department: Immunology and Immunotherapy

title-block-banner: "#f0f3f5"
title-block-banner-color: "black"

keywords: "R, Data Analysis, Data Viz"
date: today

format: 
  html: 
    toc: true
    number-sections: true
    code-overflow: wrap
    
fig-cap-location: margin
# execute: 
#  warning: false
editor: visual
---

# Intro

```{r}
#| warning: false

# Load packages

library(maftools) # For majority of maf file analysis
library(mclust)
library("BSgenome.Hsapiens.UCSC.hg38", quietly = TRUE)
library(NMF) # For signature calculation
library(pheatmap) # For nice heatmaps
library(ghibli) # For pretty colors
```

## Loading in the data

Load in raw MAF file and the clinical data

```{r}
laml = read.maf(maf="input/merged_consensus_all_samples_germline.maf", 
                clinicalData ="input/NRG-GY00_laml_annot_2.csv",
                rmFlags = TRUE # "FLAGS, frequently mutated genes in public exomes"
                )
```

::: callout-tip
Setting `rmFlags` = `TRUE` removes **FLAGs**: frequently mutated genes in public exomes

Details on these genes can be found [here](https://bmcmedgenomics.biomedcentral.com/articles/10.1186/s12920-017-0309-7#MOESM1).
:::

## Subsetting datasets

We have some normal samples with no matched tumor, and we may be interested in doing subtype-specific analysis.

Let's subset to only tumor samples, and then create further subsetted datasets based on clinical characteristics.

### Only tumor samples

I also annotated with pathogenic germline variants; the only relevant ones were in BRCA1 and BRCA2, so we are subsetting to somatic mutations and pathogenic germline variants.

```{r}
laml_tum = subsetMaf(maf = laml, clinQuery = "Tissue %in% 'Tu'")
laml_tum = subsetMaf(maf = laml_tum, query = "vcf_id %in% c('SOMATIC','GERMLINE_PATHOGENIC') ")
```

### Other subsets

We can also subset by tumor histology, treatment, and response.

:::{.callout-note collapse="true" appearance="simple"}
#### See other subsets

```{r}
## Comparing Serous vs others
laml_ser = subsetMaf(maf = laml_tum, clinQuery = "celltype %in% 'Serous_Adenocarcinoma'")
# Subsetting the other dataset to "not serous"
`%ni%` <- Negate(`%in%`)
laml_other = subsetMaf(maf = laml_tum, clinQuery = "celltype %ni% 'Serous_Adenocarcinoma'")
## Comparing CB vs NCB
laml_CB = subsetMaf(maf = laml_tum, clinQuery = "responseCB %in% 'CB'")
laml_NCB = subsetMaf(maf = laml_tum, clinQuery = "responseCB %in% 'NCB'")
## Subsetting by treatment to see if there are differences
laml_nivo = subsetMaf(maf = laml_tum, clinQuery = "tx %in% 'Nivo'")
laml_combo = subsetMaf(maf = laml_tum, clinQuery = "tx %in% 'Combo'")
```
:::

## View the MAF file summary
```{r}
#Typing laml shows basic summary of MAF file.
laml_tum

# These are some more summaries you can generate
# I'm commenting them out as they have too much info for the tables to be readable.
# #Shows sample summary.
# getSampleSummary(laml_tum)
# #Shows gene summary.
# getGeneSummary(laml_tum)
# #shows clinical data associated with samples
# getClinicalData(laml_tum)
# #Shows all fields in MAF
# getFields(laml_tum)
```

# Visualization
## Summarize the maf file

We can use `plotmafSummary` to plot the summary of the maf file, which displays number of variants in each sample as a stacked barplot and variant types as a boxplot summarized by `Variant_Classification`.

```{r}
plotmafSummary(maf = laml_tum, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```

```{r}
#Use mafbarplot for a minimal barplot of mutated genes.
mafbarplot(maf =  laml_tum)
```

## Summarize the other subsets

Here are the mutation distributions for the other subsets.

As a sanity check, there are few TP53 mutations in the top 10 for the non-Serous samples, which makes sense.

:::{.callout-note collapse="true"}

### See the other subsets

#### By Histology
##### Serous
```{r}
## Serous 
plotmafSummary(maf = laml_ser, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```
```{r}
mafbarplot(maf =  laml_ser)
```

##### Non-Serous
```{r}
## Non-serous
plotmafSummary(maf = laml_other, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```
```{r}
mafbarplot(maf =  laml_other)
```


#### By Response
##### CB
```{r}
## CB
plotmafSummary(maf = laml_CB, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```
```{r}
mafbarplot(maf =  laml_CB)
```

##### NCB
```{r}
## NCB
plotmafSummary(maf = laml_NCB, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```
```{r}
mafbarplot(maf =  laml_NCB)
```

#### By Treatment
##### Nivo
```{r}
## Nivo
plotmafSummary(maf = laml_nivo, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```
```{r}
mafbarplot(maf =  laml_nivo)
```
##### Combo
```{r}
## Combo
plotmafSummary(maf = laml_combo, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)
```
```{r}
mafbarplot(maf =  laml_combo)
```

:::

# Oncoplots

```{r}
#oncoplot for top ten mutated genes.
oncoplot(maf = laml_tum, top = 10)

# Highlight by specific attribute in MAF file
oncoplot(maf = laml_tum, 
         additionalFeature = c("vcf_id", "GERMLINE_PATHOGENIC"))
```
